crates/hir/src/lib.rs

```diff
@@ -661,8 +662,51 @@ impl Module {
                 _ => (),
             };
 
-            for item in impl_def.items(db) {
-                let def: DefWithBody = match item {
+            if let Some(trait_) = trait_ {
+                let items = &db.trait_data(trait_.into()).items;
+                let required_items = items.iter().filter(|&(_, assoc)| match *assoc {
+                    AssocItemId::FunctionId(it) => !db.function_data(it).has_body(),
+                    AssocItemId::ConstId(_) => true,
+                    AssocItemId::TypeAliasId(it) => db.type_alias_data(it).type_ref.is_none(),
+                });
+                impl_assoc_items_scratch.extend(db.impl_data(impl_def.id).items.iter().map(
+                    |&item| {
+                        (
+                            item,
+                            match item {
+                                AssocItemId::FunctionId(it) => db.function_data(it).name.clone(),
+                                AssocItemId::ConstId(it) => {
+                                    db.const_data(it).name.as_ref().unwrap().clone()
```

Comment:
Just realized I forgot to replace the unwrap here. @Young-Flash since you are looking into https://github.com/rust-lang/rust-analyzer/issues/15909 already, do you mind changing this to `?` and make this `map` a `filter_map`?

Reply:
sure